test/e2e/tuning_options_test.go

```diff
@@ -320,3 +327,195 @@ func TestHAProxyTimeoutsRejection(t *testing.T) {
 		}
 	}
 }
+
+// TestCookieLen verifies that when logging is enabled and a cookie capture is specified, any captured cookies are
+// truncated at the expected max length.
+func TestCookieLen(t *testing.T) {
+	testCases := []struct {
+		name      string
+		maxLength int
+	}{
+		{"63 bytes", 63},
+		{"64 bytes", 64},
+		{"128 bytes", 128},
+		{"1024 bytes", 1024},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			maxLength := testCase.maxLength
+			t.Parallel()
+			icName := types.NamespacedName{
+				Namespace: operatorNamespace,
+				Name:      names.SimpleNameGenerator.GenerateName("test-cookielen-"),
+			}
+			domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+			ic := newPrivateController(icName, domain)
+			cookieName := "X-foo-bar"
+			ic.Spec.Logging = &operatorv1.IngressControllerLogging{
+				Access: &operatorv1.AccessLogging{
+					Destination: operatorv1.LoggingDestination{
+						Type: "Container",
+						Container: &operatorv1.ContainerLoggingDestinationParameters{
+							MaxLength: 2048,
+						},
+					},
+					HTTPCaptureCookies: []operatorv1.IngressControllerCaptureHTTPCookie{{
+						IngressControllerCaptureHTTPCookieUnion: operatorv1.IngressControllerCaptureHTTPCookieUnion{
+							MatchType: operatorv1.CookieMatchTypeExact,
+							Name:      cookieName,
+						},
+						MaxLength: maxLength,
+					}},
+					HttpLogFormat: "cookie: %CC",
+				},
+			}
+			if err := kclient.Create(context.TODO(), ic); err != nil {
+				t.Fatalf("Failed to create ingresscontroller %s: %v", icName, err)
+			}
+			defer assertIngressControllerDeleted(t, kclient, ic)
+
+			if err := waitForIngressControllerCondition(t, kclient, 2*time.Minute, icName, availableConditionsForPrivateIngressController...); err != nil {
+				t.Fatalf("Timed out waiting for ingresscontroller %s to become available: %v", icName.Name, err)
+			}
+
+			// Send a request through the router that includes the captured cookie, and verify that it's truncated as expected.
+
+			// Use the router image for the client pod since it includes curl.
+			routerDeployment := &appsv1.Deployment{}
+			routerDeploymentName := controller.RouterDeploymentName(ic)
+			if err := kclient.Get(context.TODO(), routerDeploymentName, routerDeployment); err != nil {
+				t.Fatalf("Failed to get router deployment for ingresscontroller %s: %v", icName.Name, err)
+			}
+
+			curlPodImage := ""
+			for _, container := range routerDeployment.Spec.Template.Spec.Containers {
+				if container.Name == "router" {
+					curlPodImage = container.Image
+					break
+				}
+			}
+			if curlPodImage == "" {
+				t.Fatalf("Failed to find container \"router\" in deployment %s", routerDeployment.Name)
+			}
+			curlPodName := types.NamespacedName{
+				Name:      names.SimpleNameGenerator.GenerateName("cookielen-curl-"),
+				Namespace: "default",
+			}
+			curlPod := buildExecPod(curlPodName.Name, curlPodName.Namespace, curlPodImage)
```

Comment:
Why not use `buildCurlPod`?

Reply:
Good question. `buildCurlPod` does look like it would simplify things; I'll rework it to use that.